src/gtk/toga_gtk/widgets/tree.py

```diff
@@ -27,50 +27,22 @@ def create(self):
         self.native.set_min_content_height(200)
 
     def row_data(self, item):
-        return [
+        return [item] + [
             str(getattr(item, attr))
             for attr in self.interface._accessors
         ]
 
-    # Gtk.TreeIter cannot be compared with __eq__ !!!
-    def compare_tree_iters(self, one, two):
-        return self.store.get_path(one) == self.store.get_path(two)
-
     # TODO: Remove this function once a consistent API exists for
     # checking if a TreeSource OR Node has children.
     def node_has_children(self, parent):
         return (
             hasattr(parent, '_children') and parent._children) or (
             hasattr(parent, '_roots') and parent._roots)
 
-    def get_node(self, tree_iter):
-        def iterate_tree(node, root=False):
-            yield node
-            if self.node_has_children(node):
-                for child in node:
-                    yield from iterate_tree(child)
-
-        for node in iterate_tree(self.interface.data, root=True):
-            if self.get_impl(node) and tree_iter:
-                if self.compare_tree_iters(tree_iter, self.get_impl(node)):
-                    return node
-
-    def set_impl(self, node, impl):
-        node._impl = getattr(node, '_impl', {})
-        node._impl[self] = impl
-
-    def del_impl(self, node):
-        if hasattr(node, '_impl'):
-            del node._impl[self]
-
-    def get_impl(self, node):
-        if hasattr(node, '_impl'):
-            return node._impl.get(self, None)
-
```

Comment:
I like that you dissolved these methods -- I had forgotten that these checks are only needed once when the row is first inserted.

Reply:
Yeah. Same goes for all the get/set/delete methods. `_impl` *must* exist after it's been added to a tree - otherwise there's a big problem. So you don't need to protection against non-existence of the `_impl` attribute. And there's only one way to add to a tree.